test/e2e/step-definitions/generic-steps.ts

```diff
@@ -0,0 +1,77 @@
+import { browser, WebDriver, element, by } from 'protractor';
+
+const { Given, Then, When, setDefaultTimeout, After, Status } = require('cucumber');
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+
+setDefaultTimeout(10 * 1000);
+
+Given('I am a mobile app user', function() {
+  // Some sort of authentication setup
+});
+
+When('I launch the mobile app', function() {
+  return browser.get('');
+});
+
+Then('I should see the login screen', function() {
+  return expect(
+    element
+      .all(by.css('ion-navbar:first-child'))
+      .first()
+      .getText()
+  ).to.eventually.equal('DVSA');
+});
+
+Given('I am on the login page', function() {
+  return browser.get('');
+});
+
+When('I click the {string} button', function(buttonName) {
+  var buttonElement = element(by.xpath('//button[span[text()="' + buttonName + '"]]'));
+  return buttonElement.click();
+});
+
+Then('I successfully log in and see the journal page', function() {
+  return expect(
+    element
+      .all(by.css('span.mes-header-md'))
+      .first()
+      .getText()
+  ).to.eventually.equal('Your Journal');
+});
+
+Given('I am logged in and on the journal page', function() {
+  var loginButton = element(by.xpath('//button[span[text()="Ok"]]'));
```

Comment:
Perhaps using IDs would be a better idea? That way if do a simple text change the tests don't break?

Reply:
If the intention is to rework all the elements with ID's then yes the use of ID's would be preferable